modules/ROOT/pages/dataweave-extension-plugin.adoc

```diff
@@ -0,0 +1,397 @@
+
+//image::animations/new_project.gif[New project creation steps]
+//![](animations/new_project.gif)
+
+When creating a DataWeave project, the DataWeave 2.0 extension requires all dependencies to be loaded and indexed before it can search the libraries and make them available in your project. Visual Studio Code provides a notification bar at the bottom of the screen editor that shows when the dependencies are loading and when the indexing is complete.
+
+image::dataweave-extension-dependencies.png[Dependencies loading and indexing]
+//image::animations/loading.gif[Dependencies loading and indexing]
+// ![](animations/loading.gif)
+
+== Create a DataWeave Module
+
+DataWeave modules are `.dwl` files that define functions, variables, types, and namespaces. These files cannot contain output directives, body expressions, or `---` separators between header and body sections. A DataWeave module enables you to share all the content defined in the `.dwl` file for reuse.
+
+To create a new DataWeave module:
+
+. In Visual Studio Code, click *View > Command Palette*.
+. In the search bar, type `dataweave`.
+. Select *DataWeave: Create New Module*.
++
+
+image::dataweave-extension-createmodule.png[Command Palette with DataWeave: Create new module option selected]
+//image::animations/new_mapping.gif[Creating new Module using DataWeave Extension Plugin]
+// ![](animations/new_mapping.gif)
+
+== Test a DataWeave Module
+
+There are two ways to test your module:
+
+* By running a unit test
+* By creating a test mapping located in your `src/test/dw` folder to avoid exposing the test to to consumers of your module
+
+The tests use functions of the modules `dw::test::Tests` and `dw::test::Asserts` provided by the artifact `org.mule.weave.wtf` of the
+//xref:dataweave-testingframework.adoc[DataWeave Testing Framework].
+
+To generate a unit test for a DataWeave module, follow these steps:
+
+. In Visual Studio Code, open your DataWeave project.
+. Navigate to the `src/main/dw` folder.
+. Select your module `.dwl` file.
+. In your module `.dwl` file, click the *Add Unit Test* button located above your function.
++
+image::dataweave-extension-testmodule-1.png[MyModule.dwl file showing the function and the Add Unit Test button]
+
+//image::animations/moduleTest.gif[Test Module Steps]
+// ![](animations/moduleTest.gif)
+
+This behavior generates:
+
+* A new `.dwl` test file for the module under the `src/test/dw` folder. This file is where test cases and test suites are described:
+** A test case is a single test with the evaluation and assertion of the result.
+** A test suite groups tests with a name that better describes the intention of some related cases.
++
+The test file defines a test case for the function that you call with specific arguments and use to make assertions over the output.
+
+* Either a new test suite for the function, consisting of your new test case, or a new test case to an existing test suite.
```

Comment:
Could you re-review this section @dukesphere @martincousido when you have a moment? Seem like a tongue twister...

Reply:
@kevintroller done, feel free to tweak if you need to